orbit/pkg/setup_experience/setup_experience.go

```diff
@@ -13,28 +22,265 @@ type Client interface {
 // SetupExperiencer is the type that manages the Fleet setup experience flow during macOS Setup
 // Assistant. It uses swiftDialog as a UI for showing the status of software installations and
 // script execution that are configured to run before the user has full access to the device.
+// If the setup experience is supposed to run, it will launch a single swiftDialog instance and then
+// update that instance based on the results from the /orbit/setup_experience/status endpoint.
 type SetupExperiencer struct {
 	OrbitClient Client
+	closeChan   chan struct{}
+	rootDirPath string
+	// Note: this object is not safe for concurrent use. Since the SetupExperiencer is a singleton,
+	// its Run method is called within a WaitGroup,
+	// and no other parts of Orbit need access to this field (or any other parts of the
+	// SetupExperiencer), it's OK to not protect this with a lock.
+	sd *swiftdialog.SwiftDialog
+	// Name of each step -> is that step done
+	steps   map[string]bool
+	started bool
 }
 
-func NewSetupExperiencer(client Client) *SetupExperiencer {
-	return &SetupExperiencer{OrbitClient: client}
+func NewSetupExperiencer(client Client, rootDirPath string) *SetupExperiencer {
+	return &SetupExperiencer{
+		OrbitClient: client,
+		closeChan:   make(chan struct{}),
```

Comment:
Maybe we should remove this or close it when we're done instead of sending a struct. As-is only one channel will get the message when we close the channel

Reply:
Could also wait for cleanup branch, it's not hurting anyone rn